src/middlewared/middlewared/service.py

```diff
@@ -870,6 +871,33 @@ def ping(self):
         "ping" protocol message.
         """
         return 'pong'
+    
+
+    def _ping_host(self, host, count, timeout):
+        if osc.IS_LINUX:
+            process = run(['ping', '-c', f'{count}', '-w',  f'{timeout}', host])
```

Comment:
I don't think we should be using count or at least not defaulting to 1 only ? Just timeout should be good. Would you agree @themylogin ?

Reply:
Why? `ping -c1` is how we always check availability of the host, aren't we?

Reply:
First ping request can sometimes be lost, depending on network topology/firewall.

We are also waiting some clarification o Ken about other checks, e.g. ssh/middleware

Reply:
We are exposing a timeout option and the consumer would expect that if no connection is made we would be waiting at least timeout X seconds. If we only send 1 request and that fails, we don't wait the remaining timeout to ensure another request worked ?

Reply:
Ok, agreed